src/panel/direction/RoutesList.jsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+
+import { Item, ItemList } from 'src/components/ui/ItemList';
+import PlaceholderText from 'src/components/ui/PlaceholderText';
+import Route from './Route';
+
+const RoutesList = ({
+  routes,
+  activeRouteId,
+  activeDetails,
+  origin,
+  destination,
+  vehicle,
+  toggleRouteDetails,
+  openPreview,
+  selectRoute,
+  isLoading,
+}) => {
+
+  return isLoading
+    ? <Placeholder />
+    : <ItemList>
+      {routes.map(route =>
+        <Item
+          key={route.id}>
+          <Route
+            id={route.id}
+            route={route}
+            origin={origin}
+            destination={destination}
+            vehicle={vehicle}
+            isActive={route.id === activeRouteId}
+            showDetails={route.id === activeRouteId && activeDetails}
```

Comment:
Please note we use route.id as identifier instead of index. This is more robust and required for reordering routes on mobile.